elyra/pipeline/validate.py

```diff
@@ -293,23 +293,26 @@ async def _validate_node_properties(self, pipeline: dict, pipeline_runtime: str,
 
                     # Validate runtime components against specific node properties in component registry
                     else:
-                        property_list = await self._get_component_properties(pipeline_runtime, components, node['op'])
-                        cleaned_property_list = list(map(lambda x: str(x).replace('elyra_', ''), property_list.keys()))
+                        # This is the full dict of properties for the operation e.g. current params, optionals etc
+                        property_dict = await self._get_component_properties(pipeline_runtime, components, node['op'])
+                        cleaned_property_list = list(map(lambda x: str(x).replace('elyra_', ''),
+                                                         property_dict['current_parameters'].keys()))
 
                         # Remove the non component_parameter jinja templated values we do not check against
                         cleaned_property_list.remove('component_source')
                         cleaned_property_list.remove('label')
 
                         for node_property in cleaned_property_list:
                             if node_property not in list(node_data.keys()):
-                                response.add_message(severity=ValidationSeverity.Error,
-                                                     message_type="invalidNodeProperty",
-                                                     message="Node is missing field",
-                                                     data={"nodeID": node['id'],
-                                                           "nodeName": node_label,
-                                                           "propertyName": node_property})
+                                if self._is_required_property(property_dict, f"elyra_{node_property}"):
+                                    response.add_message(severity=ValidationSeverity.Error,
+                                                         message_type="invalidNodeProperty",
+                                                         message="Node is missing field",
```

Comment:
I think it would be good to indicate this field is required: "Node is missing required field".

Is there a reason "field" is used in these messages rather than "property" or "parameter".  I feel like we already use these two terms interchangeably and this is adding a third variant.  Given the title on the UI for editing these is "Node properties", I think "property" may be the more correct term here.

Reply:
I agree